File: JAVA/1598. Crawler Log Folder.java
class Solution {
    public int minOperations(String[] logs) {
        int res = 0;
        for (String log: logs) {
            res += add(log,res);
        }
        return res;
    }
    private int add(String log, int res) {
        return log.charAt(1) == '.' ? res == 0 ? 0 : -1 : log.charAt(0) == '.' ? 0 : 1;
    }
}

File: C++/1598. Crawler Log Folder.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int level=0;
        for(string log:logs){
            if(log=="../"){
                if(level!=0){
                    level--;
                }
            }
            else if(log=="./"){
                continue;
            }
            else{
                level++;
            }
        }
        return level;
    }
};
